test-suite/testkit/tests/interfaces/services.rs

```diff
@@ -222,3 +222,64 @@ impl DefaultInstance for AnyCallService {
     const INSTANCE_ID: u32 = Self::ID;
     const INSTANCE_NAME: &'static str = "any-call";
 }
+
+#[exonum_interface(auto_ids)]
+pub trait CustomCallInterface<Ctx> {
+    type Output;
+    fn custom_call(&self, context: Ctx, arg: Vec<u8>) -> Self::Output;
+}
+
+type CustomCall = Box<dyn Fn(ExecutionContext<'_>) -> Result<(), ExecutionError> + Send + 'static>;
```

Comment:
Nit: Do the custom call closures capture variables? (As far as I see, they don't.) If not, then it could be possible to use `fn(...)` pointers instead of closures, which would allow to simplify code substantially (`fn` pointers are `Copy + Send + 'static` by design).

Reply:
Yes, I can just use function pointer here instead of factory. But current approach is more flexible in future.

Reply:
This is test code though; I don't think future compatibility should be the primary concern 🙃

Reply:
Yes, but current code works fine and looks not so ugly.